### PR TITLE
Simplify test only project precommit checks

### PR DESCRIPTION
**Pull request by breskeby**
- we do not need thirdparty check for test only projects

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/InternalPrecommitTasks.java [hunk 1]
```diff
@@ -19,34 +19,22 @@
     /**
      * Adds a precommit task, which depends on non-test verification tasks.
      */
-    public static void create(Project project, boolean includeDependencyLicenses) {
-        project.getPluginManager().apply(JarHellPrecommitPlugin.class);
-        project.getPluginManager().apply(ThirdPartyAuditPrecommitPlugin.class);
+    public static void create(Project project, boolean withProductiveCode) {
```
Comment (pugnascotia) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/InternalPrecommitTasks.java:22]: Maybe?
```suggestion
    public static void create(Project project, boolean withProductionCode) {
```

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/InternalPrecommitTasks.java [hunk 2]
```diff
@@ -19,34 +19,22 @@
...
         project.getPluginManager().apply(LicenseHeadersPrecommitPlugin.class);
         project.getPluginManager().apply(FilePermissionsPrecommitPlugin.class);
         project.getPluginManager().apply(TestingConventionsPrecommitPlugin.class);
+        project.getPluginManager().apply(LoggerUsagePrecommitPlugin.class);
 
-        // tasks with just tests don't need dependency licenses, so this flag makes adding
+        // tasks with just tests don't need certain tasks to run, so this flag makes adding
         // the task optional
-        if (includeDependencyLicenses) {
+        if (withProductiveCode) {
```
Comment (pugnascotia) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/InternalPrecommitTasks.java:33]:
```suggestion
        if (withProductionCode) {
```

**Review by pugnascotia — APPROVED** [review]

**Comment by mark-vieira** [comment]
> JarHell check not required for test only projects

I don't necessarily think that's true. If we have duplicate classes on the testing classpath that could indicate and issue, or if nothing else, potentially cause weird test behavior/instability. I think we still want to keep that around, especially for unit and internal cluster tests.

**Comment by breskeby** [comment]
@mark-vieira hmm. makes sense as you say it. moved back the jar hell checks for all projects

**Pull request merged**